### PR TITLE
S18 t3/implement access token validation logic

### PR DESCRIPTION
**Pull request by catai9**
**Scrum Board Ticket**: S18 T3: Implement logic for validating the access token for private endpoints

**Story:** Story 18 Update System Security 

**Description:** Implemented the logic to validate the access token provided for all private requests such as booking an appointment. 

**Testing Criteria:** Tested code by running locally and sending requests to the `/test` endpoint for valid and invalid access token values. Also ensured that all automated tests in Jest passed and had 100% file coverage.

**How Reviewers can Test Code:** 
1. Run the code locally  (navigate inside the server folder and run npm start). Send requests to the `/test` endpoint (OR any other private endpoints) with valid and invalid access token values (OR valid and invalid parameters for other endpoints) on localhost port 4000 using Postman to ensure that the response is as expected. A valid access token is XcCa92ZvOnQKZsGtOKOa and it should result in true being returned. Refer to the combinations in `authenticate.test.js` for more examples of potential manual tests.
2. Run the automated tests locally and ensure all tests pass and have 100% file coverage. To do this, navigate inside the `server` folder and run `npm test`. Please note that tests have not been setup for the front end so an error will occur if one attempts to run `npm test` in the root folder.

Please note that the test endpoint URL on Postman will be `http://localhost:4000/test`

### CODE REVIEW

File: server/models/handlers/authenticate.js [hunk 1]
```diff
@@ -20,8 +23,15 @@ async function getUserFromCredentials(email, password) {
 };
 
 // Verify the access token for each private request to improve security.
+// Returns false if the access token does not match a student nor a worker's token.
 async function isAccessTokenValid(accessToken) {
-    return true;
```
Comment (prdbondi) [inline server/models/handlers/authenticate.js:24]: I see that you removed the hard coded aspect and replaced it with logic, as per your story!

**Comment by prdbondi** [comment]
Completed test 2) 

Unable to complete test 1, see attached pictures 

image
image

**Comment by prdbondi** [comment]
Test 1 now working as expected, incorrectly entered values in Postman client.

**Comment by catai9** [comment]
> 
> 
> Test 1 now working as expected, incorrectly entered values in Postman client.

Awesome, glad to see that it works now!

**Comment by prdbondi** [comment]
Ran multiple combinations of test 1) including checking the validity of entry "XcCa92ZvOnQKZsGtOKOa". Test 1 performs as expected. Code looks good, comments are in order. I am approving.

**Review by prdbondi — APPROVED** [review]
Followed testing instructions and performance was as expected. Reviewed code, comments are in order and it is simple to follow. Great PR! You have my approval.

**Pull request merged**